File: backend/app/api/routes/predict.py
from __future__ import annotations

from fastapi import APIRouter

from backend.app.schemas.predict import PredictRequest, PredictResponse
from backend.app.services.predict_service import run_predict

router = APIRouter(tags=["predict"])


@router.post("/predict", response_model=PredictResponse)
def predict(body: PredictRequest) -> PredictResponse:
    result = run_predict(
        model_name=body.model_name,
        source=body.source,
        project_id=body.project_id,
        dataset_name=body.dataset_name,
        row_start=body.row_start,
        row_end=body.row_end,
    )
    return PredictResponse(**result)

File: frontend/src/api/predict.ts
import { apiClient } from './client'
import type { PredictResult } from './types'

export interface PredictRequest {
  model_name: string
  source: 'project_test' | 'dataset'
  project_id?: string
  dataset_name?: string
  row_start?: number
  row_end?: number
}

export async function runPredict(body: PredictRequest): Promise<PredictResult> {
  const { data } = await apiClient.post<PredictResult>('/predict', body)
  return data
}
